### PR TITLE
Feature/sample

### PR DESCRIPTION
**Pull request by bkolarov-appolica**

### CODE REVIEW

File: TabController/sample/src/main/java/com/appolica/sample/tabs/NestedFragment.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        final int fragmentIndex = getArguments().getInt(KEY_FRAGMENT_INDEX);
+        binding.setText(getString(R.string.stack_fragment_index, fragmentIndex));
+
+        binding.executePendingBindings();
```
Comment (AlexanderI) [inline TabController/sample/src/main/java/com/appolica/sample/tabs/NestedFragment.java]: There is no need to do executePendingBindings

File: TabController/sample/src/main/java/com/appolica/sample/tabs/pager/ViewPagerFragment.java [hunk 2]
```diff
@@ -39,7 +29,11 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
     public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
-        final int tabNum = getArguments().getInt(BUNDLE_NUMBER);
-        binding.setTabNum("Tab " + tabNum);
+        final PagerAdapter pagerAdapter = new PagerAdapter(getChildFragmentManager());
+
+        binding.tabLayout.setupWithViewPager(binding.viewPager);
+
+        binding.viewPager.setOffscreenPageLimit(1);
```
Comment (AlexanderI) [inline TabController/sample/src/main/java/com/appolica/sample/tabs/pager/ViewPagerFragment.java]: The default value is 1 anyway

File: TabController/sample/src/main/res/layout/fragment_stack_nested.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        android:orientation="vertical">
+
+        <TextView
+            android:id="@+id/textView"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginBottom="8dp"
+            android:layout_marginLeft="8dp"
+            android:layout_marginRight="8dp"
+            android:layout_marginTop="8dp"
```
Comment (AlexanderI) [inline TabController/sample/src/main/res/layout/fragment_stack_nested.xml:28]: Just declare margin="8dp" will read better imo

File: TabController/sample/src/main/res/layout/fragment_tab_flat.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        android:layout_height="match_parent">
+
+        <TextView
+            android:id="@+id/textView"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginBottom="8dp"
+            android:layout_marginLeft="8dp"
+            android:layout_marginRight="8dp"
+            android:layout_marginTop="8dp"
```
Comment (AlexanderI) [inline TabController/sample/src/main/res/layout/fragment_tab_flat.xml:18]: margin="8dp"

File: TabController/sample/src/main/res/layout/fragment_tab_view_pager.xml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        </android.support.design.widget.TabLayout>
+
+        <android.support.v4.view.ViewPager
+            android:id="@+id/viewPager"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            android:layout_marginBottom="0dp"
+            android:layout_marginLeft="0dp"
+            android:layout_marginRight="0dp"
+            android:layout_marginTop="0dp"
```
Comment (AlexanderI) [inline TabController/sample/src/main/res/layout/fragment_tab_view_pager.xml:33]: margin="0dp"

**Review by AlexanderI — CHANGES_REQUESTED** [review]
Couple comments

File: TabController/sample/src/main/res/layout/fragment_stack_nested.xml (hunk 3 above)
Comment (bkolarov-appolica) [reply TabController/sample/src/main/res/layout/fragment_stack_nested.xml:28]: layout_margin is ignored by the constraint layout editor.

**Comment by bkolarov-appolica** [comment]
The necessary changes are made.

**Review by AlexanderI — APPROVED** [review]

**Pull request merged**